### PR TITLE
Scan Kustomize Directory

### LINKED ISSUE #718: Scan Kustomize charts
**Description**
Currently, Kubescape scans YAMLs and Helm templates but not [kustomize](https://kubernetes.io/docs/tasks/manage-kubernetes-objects/kustomization/) charts.

**Additional context**
Kubescape should generate the Kubernetes YAMLs from the Kustomize chart and scan the files.
This should work similarly to the [helm chart](https://github.com/kubescape/kubescape/blob/master/core/pkg/resourcehandler/filesloader.go#L118) scanning mechanism

### PR DESCRIPTION
**Pull request by suhasgumma**
## Describe your changes
* Detecting if the given Directory is "Kustomize" Directory.
* Generating Kubernetes Resources(.yaml files) from the defined "kustomization.yml" file. 
* Converting them to Kubernetes Workloads for scanning using "Kubescape"

* Set the `SourceType` as Kustomize Directory 
* Set the `KustomizeDirectoryName`

## This PR fixes:
#718 

## Checklist before requesting a review

- [x] My code follows the style guidelines of this project
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have performed a self-review of my code
- [x] New and existing unit tests pass locally with my changes
- [x] If it is a core feature, I have added thorough tests.

Closes #718

)

### CODE REVIEW

**Comment by suhasgumma** [comment]
Hi @dwertent, could you please take a look at this?

**Comment by Sahocan6363** [comment]
Yanlışlıkla ödeme yapıldı iptal etmenizi istiyorum yapılan ödemeyi geri
iadesini istiyorum

10 Eyl 2022 Cmt 04:25 tarihinde Suhas Gumma ***@***.***> şunu
yazdı:

File: core/pkg/resourcehandler/filesloader.go [hunk 1]
```diff
@@ -196,6 +196,43 @@ func getResourcesFromPath(path string) (map[string]reporthandling.Source, []work
...
+					CommitterName:  commitInfo.Author.Name,
+					CommitterEmail: commitInfo.Author.Email,
+					Message:        commitInfo.Message,
+				}
+			}
+		}
+
+		workloadSource := reporthandling.Source{
+			RelativePath: source,
+			FileType:     reporthandling.SourceTypeYaml,
```
Comment (suhasgumma) [inline core/pkg/resourcehandler/filesloader.go:227]: Need another type named "SourceKustomizeDirectory". The "reporthandling" is in "opa-utils". Shall I create a Pull request there?

**Comment by dwertent** [comment]
Hi @suhasgumma , This is looking good!
Can you please provide an example of the Kustomize support?
The example should include an input Kustomize file and a screenshot of the CLI output. 
Thank you :)

**Comment by suhasgumma** [comment]
> Hi @suhasgumma , This is looking good! Can you please provide an example of the Kustomize support? The example should include an input Kustomize file and a screenshot of the CLI output. Thank you :)

` Kustomize lets you customize raw, template-free YAML files for multiple purposes, leaving the original YAML untouched and usable as is.`

So, We have to scan the original YAML files (which Kubescape already does) & check for "kustomization.yaml". If "kustomization.yaml" is present, we have to produce YAML resources and scan those as well. 

[Here's a repository covering almost all possible variations of "Kustomize" directories.](https://github.com/viadee/kustomize-examples) I will use the `prod` directory inside `overlays`  to demonstrate the modifications. 

## Before Scan Kustomize

Before Scan Kustomize



## After Scan Kustomize

After Scan Kustomize

**Comment by suhasgumma** [comment]
> `Kustomize lets you customize raw, template-free YAML files for multiple purposes, leaving the original YAML untouched and usable as is.`
> 
> So, We have to scan the original YAML files (which Kubescape already does) & check for "kustomization.yaml". If "kustomization.yaml" is present, we have to produce YAML resources and scan those as well.

@dwertent, I've updated the example. Please take a look at this.

**Comment by suhasgumma** [comment]
> @dwertent, I've updated the example. Please take a look at this.
 
Please note that these are not the exact screenshots. These are selected segments of scan results to demonstrate scanning of Kustomize Directory.

**Comment by suhasgumma** [comment]
Hi @dwertent , given it a thought and I haven't added functionality to directly scan the kustomozation file. Until now, scan would take Directory as input and generate yaml resources if "kustomization.yml" is present.

Now, after making a change, we can directly scan the "kustomization.yml" file. Of course, the files used by kustomize should be present in the path specified by the "kustomization.yml" file.

Scan Kustomize Directly

**Review by dwertent — APPROVED** [review]
Well done!

**Pull request merged**